### PR TITLE
added kms resources

### PR DESCRIPTION
**Pull request by mayank0202**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+variable "namespace" {
+  type        = string
+  description = "Namespace for the resources."
+  default     = "arc"
+}
+
+variable "environment" {
+  type        = string
+  description = "Name of the environment resources will belong to."
+  default     = "poc"
```
Comment (tsaucier-sf) [inline variables.tf:79]: Don't set a default for this variable in the module

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+variable "region" {
+  type        = string
+  default     = "us-east-1"
+  description = "AWS region"
+}
+
+variable "namespace" {
+  type        = string
+  description = "Namespace for the resources."
+  default     = "arc"
```
Comment (tsaucier-sf) [inline variables.tf:73]: Don't set a default for this variable in the module

**Review by tsaucier-sf — CHANGES_REQUESTED** [review]

**Review by tsaucier-sf — APPROVED** [review]

File: main.tf [hunk 3]
```diff
@@ -1,13 +1,21 @@
-################################################################################
-## defaults
-################################################################################
-terraform {
-  required_version = "~> 1.3, < 2.0.0"
+provider "aws" {
```
Comment (vijay-stephen) [inline main.tf:1]: Provider block should be in root module not in Child.
cc : @tsaucier-sf

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+variable "enable_key_rotation" {
+  type        = bool
+  default     = true
+  description = "Specifies whether key rotation is enabled"
+}
+
+variable "description" {
+  type        = string
+  default     = "Parameter Store KMS master key"
```
Comment (vijay-stephen) [inline variables.tf:15]: Please correct it, its not `Parameter store`

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  default     = {}
+}
+
+variable "enabled" {
+  type        = bool
+  default     = true
+  description = "Set to false to prevent the module from creating any resources"
+}
+
+variable "region" {
```
Comment (vijay-stephen) [inline variables.tf:61]: This variable is not required when provider block is moved out.

File: main.tf [hunk 6]
```diff
@@ -1,13 +1,21 @@
...
 
-  required_providers {
-    aws = {
-      source  = "hashicorp/aws"
-      version = "~> 4.0"
-    }
-  }
+resource "aws_kms_alias" "default" {
+  count         = var.enabled ? 1 : 0
+  name          = coalesce(var.alias, format("alias/%v", var.tags))
```
Comment (vijay-stephen) [inline main.tf:19]: Format is outdated, please use latest format

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+variable "description" {
+  type        = string
+  default     = "Parameter Store KMS master key"
+  description = "The description of the key as viewed in AWS console"
+}
+
+variable "alias" {
+  type        = string
+  default     = "alias/eks-kms-key"
```
Comment (vijay-stephen) [inline variables.tf:21]: default is not required.

File: example/main.tf [hunk 8]
```diff
@@ -1,17 +1,40 @@
...
+
+
+module "kms" {
+  source                  = "../" # Use the local path to your kms-module directory
+  enabled                 = var.enabled
+  deletion_window_in_days = var.deletion_window_in_days
+  enable_key_rotation     = var.enable_key_rotation
+  alias                   = var.alias
+  tags                    = module.tags.tags
+  policy                  = var.policy
```
Comment (vijay-stephen) [inline example/main.tf:39]: Where are you defining policy ?

Comment (mayank0202) [reply example/main.tf:39]: I haven't defined the policy in example folder, just specified the empty variable for the policy, i will add a policy in example folder

File: main.tf (hunk 3 above)
Comment (mayank0202) [reply main.tf]: Done

File: variables.tf (hunk 4 above)
Comment (mayank0202) [reply variables.tf]: Done

File: variables.tf (hunk 5 above)
Comment (mayank0202) [reply variables.tf]: Done

File: variables.tf (hunk 7 above)
Comment (mayank0202) [reply variables.tf]: Done

File: example/main.tf (hunk 8 above)
Comment (mayank0202) [reply example/main.tf:39]: Its added in `examplevariables.tf` now

**Review by tsaucier-sf — APPROVED** [review]

**Pull request merged**